oracle/src/main.rs

```diff
@@ -211,71 +135,42 @@ async fn main() -> Result<(), Error> {
     );
     let opts: Opts = Opts::parse();
 
-    log::info!("Starting oracle with currencies = {:?}", opts.currency_id);
+    // read price configs from file
+    let data = std::fs::read_to_string(opts.oracle_config)?;
+    let oracle_config = serde_json::from_str::<OracleConfig>(&data)?;
+    // validate routes
+    for price_config in &oracle_config.prices {
+        price_config.validate().map_err(Error::InvalidConfig)?
+    }
+
+    let currency_store = &oracle_config.currencies;
+    let mut price_feeds = feeds::PriceFeeds::new(currency_store.clone());
+    price_feeds.add_coingecko(opts.coingecko);
```

Comment:
A bit misleading since these functions have `if let Some(x)` inside them. Either this should be something like `maybe_add_coingecko`, or we do the `if let` here